packages/webcert/src/components/header/WebcertHeaderUnit.tsx

```diff
@@ -24,25 +36,104 @@ const InactiveUnit = styled.span`
   font-size: 12px;
 `
 
+const StyledButton = styled.button`
+  border: none;
+`
+
+const Link = styled.button`
+  text-decoration: none;
+`
+
+const ExpandedDiv = styled.div`
+  position: absolute;
+  background: #ffffff;
+  padding: 1.2em 2em;
+  box-shadow: 1px 1px 10px #ccc;
+  right: -11px;
+  margin-top: 0.5em;
+  border-radius: 3px;
+  z-index: 9999;
+
+  &::before {
+    content: '';
+    width: 10px;
+    height: 10px;
+    background: #ffffff;
+    transform: rotate(45deg);
+    position: absolute;
+    top: -5px;
+    right: 12px;
+  }
+`
+
+const RelativeDiv = styled.div`
+  position: relative;
+`
+
 const WebcertHeaderUnit: React.FC = () => {
+  const dispatch = useDispatch()
   const user = useSelector(getUser, shallowEqual)
+  const totalDraftsAndUnhandledQuestionsOnOtherUnits = useSelector(getTotalDraftsAndUnhandledQuestionsOnOtherUnits)
+  const userLinks = useSelector(getUserResourceLinks)
+  const [isExpanded, setIsExpanded] = useState(false)
+
+  const changeUnitLink = userLinks?.find((link) => link.type === ResourceLinkType.CHANGE_UNIT)
+
+  const toggleMenu = () => {
+    setIsExpanded(!isExpanded)
+  }
+
+  const expandButton = () => {
```

Comment:
This should be a general component. Maybe there is a component that you could have reused.

Reply:
Tests are missing for closing button, can only find test to open